### PR TITLE
Updated amend Raml

### PR DESCRIPTION
**Pull request by BackRowOla**

### CODE REVIEW

File: resources/public/api/conf/1.0/examples/AmendCisDeductions/amend_cis_deductions_request.json [hunk 1]
```diff
@@ -1,8 +1,4 @@
 {
-  "fromDate": "2020-01-01",
-  "toDate": "2020-02-01",
-  "contractorName": "contractor1",
-  "employeeRef": "135/AB56789",
   "periodData": [
     {
       "deductionAmount": 1025.10,
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/examples/AmendCisDeductions/amend_cis_deductions_request.json:4]: Can you use the example from the Tech Spec with the multiple periods

File: resources/public/api/conf/1.0/schemas/amendCisDeductions/amend_cis_deductions_request.json [hunk 2]
```diff
@@ -36,41 +16,35 @@
             "example": "5000.99"
           },
           "deductionFromDate": {
-            "description": "The From date when the deduction was made",
+            "description": "The From date when the deduction was made within the CIS tax year period.",
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/schemas/amendCisDeductions/amend_cis_deductions_request.json:19]: Not at this line but the deductionAmount (above) description is not correct

File: resources/public/api/conf/1.0/schemas/amendCisDeductions/amend_cis_deductions_request.json [hunk 3]
```diff
@@ -36,41 +16,35 @@
...
             "type": "string",
             "example": "2019-04-04"
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/schemas/amendCisDeductions/amend_cis_deductions_request.json:21]: Can you make the deductionFromDate the 6th of a month and the deductionToDate the 5th of the following month

File: resources/public/api/conf/1.0/scenarios/amend_scenarios.md [hunk 4]
```diff
@@ -18,38 +18,22 @@
...
         <tr>
             <td><p>RULE_DEDUCTIONS_DATE_RANGE_INVALID</p></td>
             <td><p>Simulate the scenario where the deductions date range is longer than 366 or less than a day</p></td>
         </tr>
         <tr>
-            <td><p>RANGE_DEDUCTIONS_TO_DATE_BEFORE_DEDUCTIONS_FROM_DATE</p></td>
-            <td><p>Simulate the scenario where the deductions To date must be after the Deductions From date</p></td>
-        </tr>                
+            <td><p>MATCHING_RESOURCE_NOT_FOUND</p></td>
+            <td><p>Simulate the scenario where The remote endpoint has indicated that no data can be found for the given period</p></td>
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/scenarios/amend_scenarios.md:28]: Change the capital letters in the middle of the sentence to lowercase.

File: resources/public/api/conf/1.0/amend.raml [hunk 5]
```diff
@@ -7,17 +7,10 @@ is:
   - errors.formatDeductionFromDate
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:7]: The wording on this has changed

File: resources/public/api/conf/1.0/amend.raml [hunk 6]
```diff
@@ -7,17 +7,10 @@ is:
...
   - errors.formatDeductionToDate
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:8]: The wording on this has changed

File: resources/public/api/conf/1.0/amend.raml [hunk 7]
```diff
@@ -7,17 +7,10 @@ is:
...
   - errors.ruleDeductionAmount
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:9]: The wording on this has changed

File: resources/public/api/conf/1.0/amend.raml [hunk 8]
```diff
@@ -7,17 +7,10 @@ is:
...
-  - errors.formatFromDate
-  - errors.formatToDate
   - errors.ruleCostOfMaterials
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:10]: The wording on this has changed

File: resources/public/api/conf/1.0/amend.raml [hunk 9]
```diff
@@ -7,17 +7,10 @@ is:
...
   - errors.ruleGrossAmountPaid
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:11]: The wording on this has changed

File: resources/public/api/conf/1.0/amend.raml [hunk 10]
```diff
@@ -7,17 +7,10 @@ is:
...
-  - errors.ruleDateRangeInvalid
   - errors.ruleDeductionsDateRangeInvalid
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:12]: The wording on this has changed

File: resources/public/api/conf/1.0/amend.raml [hunk 11]
```diff
@@ -7,17 +7,10 @@ is:
...
   - errors.rangeDeductionToDateBeforeDeductionsFromDate
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:13]: In fact I think you need to align all the errors with the wording changes and code changes that have been made in the business spec

File: resources/public/api/conf/1.0/errors.raml [hunk 12]
```diff
@@ -118,7 +118,7 @@ traits:
             type: types.errorResponse
             examples:
               formatDeductionFromDate:
-                description: The deductions From date is not a valid ISO format date
+                description: The deductions From date is not a valid date format
                 value:
                   code: FORMAT_DEDUCTION_FROM_DATE
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/errors.raml:123]: The code has changed too

File: resources/public/api/conf/1.0/errors.raml [hunk 13]
```diff
@@ -129,7 +129,7 @@ traits:
             type: types.errorResponse
             examples:
               formatDeductionToDate:
-                description: The deductions To date is not a valid ISO format date
+                description: The deductions To date is not a valid date format
                 value:
                   code: FORMAT_DEDUCTION_TO_DATE
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/errors.raml:134]: The code has changed too

File: resources/public/api/conf/1.0/errors.raml [hunk 14]
```diff
@@ -129,7 +129,7 @@ traits:
...
   ruleDeductionAmount:
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/errors.raml:135]: The description has changed

File: resources/public/api/conf/1.0/errors.raml [hunk 15]
```diff
@@ -206,7 +206,7 @@ traits:
             type: types.errorResponse
             examples:
               ruleDeductionsDateRangeInvalid:
-                description: The deductions date range is longer than 366 or less than a day
+                description: The deductions period must align from the 6th of one month to the 5th of the following month
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/errors.raml:209]: This is the new description not the one we took into the sprint.
Note the return status for this error is 403.

File: resources/public/api/conf/1.0/amend.raml [hunk 16]
```diff
@@ -7,17 +7,9 @@ is:
...
   - errors.ruleCostOfMaterials
   - errors.ruleGrossAmountPaid
-  - errors.ruleDateRangeInvalid
   - errors.ruleDeductionsDateRangeInvalid
-  - errors.rangeDeductionToDateBeforeDeductionsFromDate
-  - errors.clientOrAgentNotAuthorised
-  - errors.ruleNoChangeLossAmount
-  - errors.ruleFromDate
-  - errors.ruleToDate
   - errors.formatSubmissionId
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/amend.raml:13]: You are missing 3 error from the error table on the business spec

File: resources/public/api/conf/1.0/scenarios/amend_scenarios.md [hunk 17]
```diff
@@ -18,38 +18,22 @@
...
-            <td><p>FORMAT_FROM_DATE</p></td>
-            <td><p>Simulate the scenario where the From date is not a valid ISO format date</p></td>
-        </tr>
-        <tr>
-            <td><p>FORMAT_TO_DATE</p></td>
-            <td><p>Simulate the scenario where the To date is not a valid ISO format date</p></td>
         </tr>
         <tr>
             <td><p>RULE_DEDUCTIONS_DATE_RANGE_INVALID</p></td>
             <td><p>Simulate the scenario where the deductions date range is longer than 366 or less than a day</p></td>
```
Comment (ian-ralphs) [inline resources/public/api/conf/1.0/scenarios/amend_scenarios.md:24]: This is not the scenario for this error

File: resources/public/api/conf/1.0/errors.raml [hunk 18]
```diff
@@ -129,9 +129,9 @@ traits:
...
             examples:
               formatDeductionToDate:
-                description: The deductions To date is not a valid ISO format date
+                description: The deductions To date is not a valid date format
                 value:
-                  code: FORMAT_DEDUCTION_TO_DATE
+                  code: FORMAT_DEDUCTIONS_TO_DATE
   ruleDeductionAmount:
     responses:
       400:
```
Comment (zachhall1234) [inline resources/public/api/conf/1.0/errors.raml:137]: Line 143: I think the description should have 'deductions' not 'deduction' for RULE_DEDUCTIONS_AMOUNT

File: resources/public/api/conf/1.0/errors.raml [hunk 19]
```diff
@@ -110,36 +110,36 @@ traits:
                 description: An empty or non-matching body was submitted
                 value:
                   code: RULE_INCORRECT_OR_EMPTY_BODY_SUBMITTED
-  formatDeductionFromDate:
```
Comment (zachhall1234) [inline resources/public/api/conf/1.0/errors.raml:113]: I think these changes to the error names need to be changed anywhere else they're referenced not just Amend.raml

**Pull request merged**